pkg/solana/chainwriter/chain_writer.go

```diff
@@ -220,6 +225,76 @@ func (s *SolanaChainWriterService) FilterLookupTableAddresses(
 	return filteredLookupTables
 }
 
+// CreateATAs first checks if a specified location exists, then checks if the accounts derived from the
+// ATALookups in the ChainWriter's configuration exist on-chain and creates them if they do not.
+func CreateATAs(ctx context.Context, args any, lookups []ATALookup, derivedTableMap map[string]map[string][]*solana.AccountMeta, client client.MultiClient, idl string, feePayer solana.PublicKey) ([]solana.Instruction, error) {
+	createATAInstructions := []solana.Instruction{}
+	for _, lookup := range lookups {
+		// Check if location exists
+		if lookup.Location != "" {
+			// TODO refactor GetValuesAtLocation to not return an error if the field doesn't exist
+			_, err := GetValuesAtLocation(args, lookup.Location)
+			if err != nil {
+				// field doesn't exist, so ignore ATA creation
+				if errors.Is(err, errFieldNotFound) {
+					continue
+				}
+				return nil, fmt.Errorf("error getting values at location: %w", err)
+			}
+		}
+		walletAddresses, err := GetAddresses(ctx, args, []Lookup{lookup.WalletAddress}, derivedTableMap, client)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving wallet address: %w", err)
+		}
+		if len(walletAddresses) != 1 {
+			return nil, fmt.Errorf("expected exactly one wallet address, got %d", len(walletAddresses))
+		}
+		wallet := walletAddresses[0].PublicKey
+
+		tokenPrograms, err := GetAddresses(ctx, args, []Lookup{lookup.TokenProgram}, derivedTableMap, client)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving token program address: %w", err)
+		}
+
+		mints, err := GetAddresses(ctx, args, []Lookup{lookup.MintAddress}, derivedTableMap, client)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving mint address: %w", err)
+		}
+		if len(tokenPrograms) != len(mints) {
+			return nil, fmt.Errorf("expected equal number of token programs and mints, got %d tokenPrograms and %d mints", len(tokenPrograms), len(mints))
+		}
+
+		for i := range tokenPrograms {
+			tokenProgram := tokenPrograms[i].PublicKey
+			mint := mints[i].PublicKey
+
+			ataAddress, _, err := tokens.FindAssociatedTokenAddress(tokenProgram, mint, wallet)
+			if err != nil {
+				return nil, fmt.Errorf("error deriving ATA: %w", err)
+			}
+
+			_, err = client.GetAccountInfoWithOpts(ctx, ataAddress, &rpc.GetAccountInfoOpts{
+				Encoding:   "base64",
+				Commitment: rpc.CommitmentFinalized,
+			})
+			if err == nil {
+				continue
```

Comment:
Info log here, saying this ataAddress was already existing, so not creating.

And then I realized there is no logging anywhere in this file.
Perhaps have a separate ticket, to later on add logs in this file. Will help a lot during debugging.

Reply:
I added some logging to the ChainWriter, including in this place

Reply:
Worried Info might get too noisy since they would log on every transactions. Could we get away with Debug instead? Except maybe for the initialization log and the `Sending main transaction` one?

Reply:
Yes makes sense about noise.
We could do the other way around.
Log an info log if we are creating an ATA. Since we will less frequently create ATAs, it wouldn't be too much noise.

Reply:
Changed all logs to Debug besides the initialization and the ATA creation ones